src/java/htsjdk/samtools/BamIndexValidator.java

```diff
@@ -84,4 +85,70 @@ public static int exhaustivelyTestIndex(final SamReader reader) { // throws Exce
         return 0;
     }
 
+    /**
+     * A less time-consuming index validation that only looks at the first and last references in the index
+     *
+     * @param reader
+     * @return # of chunks examined, or 0 if there is no browseable index for the reader
+     */
+    public static int lessExhaustivelyTestIndex(final SamReader reader) {
+        // look at all chunk offsets in a linear index to make sure they are valid
+
+        if (reader.indexing().hasBrowseableIndex()) {
+
+            // content is from an existing bai file
+            final CachingBAMFileIndex existingIndex = (CachingBAMFileIndex) reader.indexing().getBrowseableIndex();
+            final int n_ref = existingIndex.getNumberOfReferences();
+
+
+            int chunkCount = 0;
+            int indexCount = 0;
+            for (int i = 0; i < n_ref; i++) {
+
+                final BAMIndexContent content = existingIndex.getQueryResults(i);
+
+                final List<Chunk> chunks = content.getAllChunks();
+                for (int j = 0; j < chunks.size(); j++) {
+
+                    chunkCount++;
+
+                    //validate the first and last chunk in each contig
+                    if (j == 0 || j==(chunks.size()-1)) {

```

Comment:
Does this even need to iterate any more?  I think you could either:
1. Check that `chunks.size()` > 0, and then just grab `chunks.get(0)` and `chunks.get(chunks.size()-1)` and test those?
2. Modify your for loop to be `for (int j = 0; j < chunks.size(); j+=chunks.size()-1)` or `for (int j : new int[]{0, chunks.size()-1})`